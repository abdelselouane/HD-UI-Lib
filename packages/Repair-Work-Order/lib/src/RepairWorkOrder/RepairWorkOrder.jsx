import 'flexboxgrid/css/flexboxgrid.min.css';
import React from 'react';
import SearchField from '../Elements/SearchField/SearchField';
import DropdownField from '../Elements/DropdownField/DropdownField';

export class RepairWorkOrder extends React.Component {
    
    state = {
        selectedItem: {
            id: 0,
            value: '',
            option: 'Work Order'
        },
        dropdownOpen: false
    }

    handleSelectedItem = (event) => {
        this.setState({
          selectedItem: {
            id: parseInt(event.target.value),
            value: event.target.id,
            option: event.target.title
          }
        });
        this.handleDropdownClose();
    }

    handleDropdownClose = () => {
        this.setState({
          dropdownOpen: false
        });
    }

    handleDropdownOpen = () => {
        this.setState({
          dropdownOpen: true
        });
    }

    render () {

        const dropdownProps = {
            id: 'dropDown',
            label: '',
            dropdownOpen: this.state.dropdownOpen,
            selectedItem: this.state.selectedItem,
            handleSelectedItem: this.handleSelectedItem,
            // handleDropdownOpen: this.handleDropdownOpen,
            items: [{
                id: 0,
                value: '',
                option: 'Work Order'
            }]
        }

        const searchFieldProps = {
            input: {
              id: 'searchField',
              type: 'text',
              placeholder: "#SSSSSAAAAAXXXXX",
              value: '',
              onChange: () => {}
            }
          };

        const searchBtn = {
            border: '1px solid #ccc',
            backgroundColor: '#fbfbfb',
            padding: '9px 15px',
            font: '18px',
            color: 'rgb(255, 120, 73)',
            width: '120px',
            minHeight: '46px',
            position: 'relative',
            marginTop: '22px',
            marginLeft: '-66px',
        }

        const searchBtnText = {
            textDecoration: 'none',
            cursor: 'pointer',
            fontSize: '18px',
            fontWeight: '500'
        }

        return (
            <React.Fragment>
                <div className="container">
                    <div className="row center-lg">
                        <div className="col-lg-12">
                            <h1 className="workOrders-text">Work Order Search</h1>
                        </div>
                    </div>
                    <div className="row center-lg">
                        <div className="col-lg-2">
                            <DropdownField {...dropdownProps} />
                        </div>
                        <div className="col-lg-6">
                            <SearchField {...searchFieldProps} />
                        </div>
                        <div className="col-lg-1">
                            <div style={searchBtn}>
                                <a style={searchBtnText}> Search </a>
                            </div>
                        </div>
                    </div>
                    <div className="row start-lg">
                        <div className="col-lg-12">
                            <br />
                            <h2>Open Work Orders</h2>
                        </div>
                    </div>
                    <div className="row end-lg">
                        <div className="col-lg-12">
                            <button className="button md primary">New Work Order</button>
                        </div>
                    </div>
                </div>
            </React.Fragment>
        );
    }
}


export default RepairWorkOrder;