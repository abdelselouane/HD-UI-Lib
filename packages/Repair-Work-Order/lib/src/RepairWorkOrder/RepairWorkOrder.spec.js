import React from 'react';
import RepairWorkOrder from './RepairWorkOrder';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as Enzyme from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new ReactSixteenAdapter() });

describe('<RepairWorkOrder />', () => {
    let wrapper;
    let state;
  
    afterEach(() => {
      jest.clearAllMocks();
    });
  
    beforeAll(() => {
      state = {
        selectedItem: {
            id: 1,
            value: 'Item1',
            option: 'Item1'
        },
        dropdownOpen: false
      };
      wrapper = shallow(<RepairWorkOrder />);
    });
  
    it('should render properly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should set selectedItem when handleSelectedItem is called', () =>{
        wrapper.setState({ dropdownOpen: true });
        const event = { target: { id: 'Item2', value: '2', title: 'Item2' }}
        wrapper.instance().handleSelectedItem(event);
        expect(wrapper.state().selectedItem).toMatchObject({ id: 2, value: 'Item2', option: 'Item2' });
        expect(wrapper.state().dropdownOpen).toBe(false);
    });

    it('should set dropdownOpen to true when handleDropdownOpen is called', () => {
        wrapper.instance().handleDropdownOpen();
        expect(wrapper.state().dropdownOpen).toBe(true);
    });
  });
  

