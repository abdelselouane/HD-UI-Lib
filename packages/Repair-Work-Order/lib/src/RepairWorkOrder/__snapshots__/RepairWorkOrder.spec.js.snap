// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepairWorkOrder /> should render properly 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="row center-lg"
    >
      <div
        className="col-lg-12"
      >
        <h1
          className="workOrders-text"
        >
          Work Order Search Test
        </h1>
      </div>
    </div>
    <div
      className="row center-lg"
    >
      <div
        className="col-lg-2"
      >
        <DropdownField
          dropdownOpen={false}
          handleSelectedItem={[Function]}
          id="dropDown"
          items={
            Array [
              Object {
                "id": 0,
                "option": "Work Order",
                "value": "",
              },
            ]
          }
          label=""
          selectedItem={
            Object {
              "id": 0,
              "option": "Work Order",
              "value": "",
            }
          }
        />
      </div>
      <div
        className="col-lg-6"
      >
        <SearchField
          input={
            Object {
              "id": "searchField",
              "onChange": [Function],
              "placeholder": "#SSSSSAAAAAXXXXX",
              "type": "text",
              "value": "",
            }
          }
        />
      </div>
      <div
        className="col-lg-1"
      >
        <div
          style={
            Object {
              "backgroundColor": "#fbfbfb",
              "border": "1px solid #ccc",
              "color": "rgb(255, 120, 73)",
              "font": "18px",
              "marginLeft": "-66px",
              "marginTop": "22px",
              "minHeight": "46px",
              "padding": "9px 15px",
              "position": "relative",
              "width": "120px",
            }
          }
        >
          <a
            style={
              Object {
                "cursor": "pointer",
                "fontSize": "18px",
                "fontWeight": "500",
                "textDecoration": "none",
              }
            }
          >
             Search 
          </a>
        </div>
      </div>
    </div>
    <div
      className="row start-lg"
    >
      <div
        className="col-lg-12"
      >
        <br />
        <h2>
          Open Work Orders
        </h2>
      </div>
    </div>
    <div
      className="row end-lg"
    >
      <div
        className="col-lg-12"
      >
        <button
          className="button md primary"
        >
          New Work Order
        </button>
      </div>
    </div>
  </div>
</Fragment>
`;
